src/blog/understanding-ecmascript-part-3.md

```diff
@@ -0,0 +1,320 @@
+---
+title: 'Understanding the ECMAScript spec, part 3'
+author: '[Marja Hölttä](https://twitter.com/marjakh), speculative specification spectator'
+avatars:
+  - marja-holtta
+date: 2020-02-11 13:33:37
+tags:
+  - ECMAScript
+description: 'Tutorial on reading the ECMAScript specification'
+tweet: ''
+---
+
+... where we dive deep in the syntax!
+
+## Previous episodes
+
+In [part 2](/blog/understanding-ecmascript-part-2), we examined a simple grammar production and how its runtime semantics are defined.  In [the extra content](/blog/extra/understanding-ecmascript-part-2-extra), we also followed a long grammar production chain from `AssignmentExpression` to `MemberExpression`. In this episode, we'll go deeper in the definition of the ECMAScript (or JavaScript) language and its syntax.
+
+If you're not familiar with [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar), now it's a good idea to check out the basics, since the spec uses context-free grammars to define the language.
```

Comment:
Offline discussion: I use the word "context", e.g., to say that "/" is division in some contexts and regexp in others. This might be confusing because of the overloaded word "context". I tried to think of a better word but couldn't find any... I'll prob add a note that "context" here doesn't mean the same as "context" in the "context-free grammar" sense.